Kernel/Arch/x86/common/ProcessorInfo.cpp

```diff
@@ -114,45 +114,66 @@ NonnullOwnPtr<KString> ProcessorInfo::build_features_string(Processor const& pro
 
 void ProcessorInfo::populate_cache_sizes()
 {
-    u32 max_extended_leaf = CPUID(0x80000000).eax();
+    // NOTE: Intel exposes detailed CPU's cache information in CPUID 04. On the
+    // other hand, AMD uses CPUID's extended function set.
+    if (m_vendor_id_string->view() == intel_vendor_id) {
```

Comment:
Let's not make a massive if/else mess out of this function, just split it into `populate_cache_sizes_amd` and `populate_cache_sizes_intel` please, calling them conditionally in the ctor :)